torch/csrc/autograd/saved_variable.cpp

```diff
@@ -131,30 +123,20 @@ Variable SavedVariable::unpack(std::shared_ptr<Node> saved_for) const {
     TORCH_CHECK(false, message.str());
   }
 
+  // The version counter is correct. If we have the original variable, we simply return it
+
   if (saved_original_) {
     return data_;
   }
 
+  // From now on, we can assume the variable is not a leaf and is an output.
```

Comment:
nit: maybe also say that we have a valid grad_fn that should be attached to the returned Tensor.